src/app/patient-search/patient-search.component.ts

```diff
@@ -69,14 +72,21 @@ export class PatientSearchComponent implements OnInit {
                             console.log(err.message);
                         },
                     })
-                    .add(() => (this.isLoading = false));
+                    .add(() => {
+                        this.isLoading = false;
+                    });
             });
     }
 
     onSelected() {
         this.selectedPatient.emit(this.searchPatientsCtrl.value);
+        this.searchedPatient(this.searchPatientsCtrl.value);
+    }
+    private searchedPatient(patient: Patient) {
+        this.route.navigate(['visit'], {
+            state: patient,
+        });
```

Comment:
this doesn't make sense. on line 82 you are doing `.emit` to notify any of the `component` that this component is a child of about that new patient has been selected, yet you added a specific code that always will reroute to the `/visit` 😕 That makes this patient search component tightly couple to the `/visit` route and that is not what we wanted. Now when you look for a patient in history and expect to be shown a a selected patient's history you get rerouted to `/visit` 

![ShareX_d1A09a9B5h](https://github.com/Johngtka/Biomagnethick-app/assets/11686968/badce806-28d8-4102-bd2f-31b3c0882bb9)

To complete this task you need to apply the same logic but in the component that we are interested in to reroute to `/visit`

Reply:
explain in points